internal/p2p/peermanager.go

```diff
@@ -101,6 +108,11 @@ type PeerManagerOptions struct {
 	// outbound). 0 means no limit.
 	MaxConnected uint16
 
+	// MaxOutgoingConnections specifies how many outgoing
+	// connections. It must be lower than MaxConnected. If it is
+	// 0, then all connections can be outgoing.
+	MaxOutgoingConnections uint16
```

Comment:
It's possible that we actually don't worry about tracking incoming and outgoing connections. We just have it if all the connections are above a certain threshold we don't bother dialing. At the end of the day the connections are the same - we just want to leave some buffer for new nodes to join

Reply:
> So if you have 200 peer slots, and 100 nodes dial you, then you'll never dial peer

That seems fine with me. The distinction this makes, other than making the implementation simpler is that it better allows us to control how many connection slots we leave open. With the current implementation. If `MaxOutgoingConnections` is 100 and 100 nodes dial us, then we'll still go ahead and try fill up all our slots which is perhaps unnecessary. By tweaking this slightly the framing is now, what is our target in terms of connections we'd like to have (alongside what is the maximum we could possibly have) rather than how many outgoing connections do we want

Reply:
> I think the total number of peers that you'll end up with will, on average, be "maxoutgoingpeers"

I don't think this will be true. As long as there are available peers out there, you will have at least "maxoutgoingpeers" not on average. 

> "why isn't tendermint being more aggressive about connecting."

You would frame things as `TargetConnections` and `MaxConnections`. Target is how much you personally care about connecting to. Max is how much you allow others to still connect to you

Reply:
I mean we could rename maxconnections (which I think would require changing the config file,) but we haven't, so I think the expectation missmatch will still exist.

Reply:
> I think the expectation missmatch will still exist.

What mismatch are you referring to?

Reply:
Ok, I think we also want to keep the incoming/outgoing peer separation: the problem we see with celestia where a node that joins a testnet late will never get enough peers to blocksync becuase nodes dial aggressively and don't leave slots for incoming peers will continue to happen. 

The incoming/outgoing tracking addresses this and is (conceptually) a forward port of an existing feature from the legacy stack.

Reply:
OK, I think that does make sense.